metals/src/main/scala/scala/meta/internal/metals/debug/DebugServer.scala

```diff
@@ -133,6 +221,42 @@ object DebugServer {
     socket
   }
 
+  private def withRebuildRetry[A](
+      f: () => Try[A],
+      compilations: Compilations,
+      buildTargets: BuildTargets,
+      buildTargetClasses: BuildTargetClasses
+  )(implicit ec: ExecutionContext): Future[A] = {
+    Future.fromTry(f()).recoverWith {
+      case _: ClassNotFoundException | _: BuildTargetNotFoundException =>
+        val allTargets = buildTargets.all.toSeq.map(_.info.getId())
+        for {
+          _ <- compilations.compileTargets(allTargets)
```

Comment:
I think even if we manage to find the build target we should still run compile on it.

So, do the same, but on single target first. If that fails, compile all targets.

Reply:
sure, added narrowed recovery for `ClassNotFoundInBuildTargetException`